myapp/tests/test_stock_api.py

```diff
@@ -10,35 +10,55 @@ def setUp(self):
         self.existed_symbols = ['AAL', 'WFC', "DFS"]
         self.not_existed_symbols = ['kamal', 'murad', 'malak']
 
+        self.valid_time_ranges = ['1d', '5d', '1m', '3m', '6m', '1y', '5y']
+        self.invalid_time_ranges = ['1x', '42z', 'rt', 'test']
+
+
     def test_service_is_up(self):
         try:
             get_stock_info(self.existed_symbols[0])
         except StockServerUnReachable as e:
             raise AssertionError(e)
 
+
     def test_get_stock_info(self):
         for symbol in self.not_existed_symbols:
             self.assertRaises(StockSymbolNotFound, get_stock_info, symbol)
         for symbol in self.existed_symbols:
             response = get_stock_info(symbol)
             self.assertIsInstance(response, dict)
 
-    def test_get_stock_historic_prices(self):
+
+    def test_get_stock_historic_prices_symbol(self):
         for symbol in self.not_existed_symbols:
             self.assertRaises(StockSymbolNotFound, get_stock_historic_prices, symbol)
         for symbol in self.existed_symbols:
             response = get_stock_historic_prices(symbol)
             self.assertIsInstance(response, list)
 
-        # testing time range
-        response = get_stock_historic_prices(self.existed_symbols[0], '1y')
-        first_year = datetime.fromisoformat(response[0].get('date')).year
-        second_year = datetime.today().year
-        self.assertEquals(second_year - first_year, 1)
-        response = get_stock_historic_prices(self.existed_symbols[0], '1m')
-        first_month = datetime.fromisoformat(response[0].get('date')).month
-        second_month = datetime.today().month
-        self.assertEquals(second_month - first_month, 1)
+
+    def test_get_stock_historic_prices_time(self):
+        for time_range in self.valid_time_ranges:
+            response = get_stock_historic_prices(self.existed_symbols[0], time_range)
+
+            if time_range.endswith('d'):
+                first_day = datetime.fromisoformat(response[0].get('date')).day
+                last_day = datetime.fromisoformat(response[-1].get('date')).day
+                self.assertEquals(last_day - first_day, int(time_range[0])-1)
```

Comment:
Does this handle the issue you mentioned about Friday and Saturday having no stocks-reports?